services/src/unit-test/java/uk/gov/moj/sdt/services/cache/ErrorMessagesCacheTest.java

```diff
@@ -49,123 +52,120 @@
  *
  * @author d130680
  */
-public class ErrorMessagesCacheTest extends AbstractSdtUnitTestBase {
-
-    /**
-     * Error messages cache.
-     */
-    private ErrorMessagesCache cache;
+@ExtendWith(MockitoExtension.class)
+class ErrorMessagesCacheTest extends AbstractSdtUnitTestBase {
 
     /**
-     * Management bean holding cache information.
+     * Generic dao.
      */
-    private ISdtManagementMBean managementMBean;
+    @Mock
+    private IGenericDao mockGenericDao;
 
     /**
-     * Generic dao.
+     * Error messages cache.
      */
-    private IGenericDao mockGenericDao;
+    private ErrorMessagesCache cache;
 
     /**
      * Dao query results.
      */
     private ErrorMessage[] result;
 
+    private static final String ERROR_DESCRIPTION_1 = "errorDescription 1";
+    private static final String ERROR_DESCRIPTION_2 = "errorDescription 2";
+    private static final String ERROR_DESCRIPTION_3 = "errorDescription 3";
+
     /**
      * Setup mock objects and inject DAO dependencies into our class.
      */
-    @Before
+    @BeforeEach
+    @Override
     public void setUp() {
+         MockitoAnnotations.openMocks(this);
+
         cache = new ErrorMessagesCache();
-        mockGenericDao = EasyMock.createMock(IGenericDao.class);
         cache.setGenericDao(mockGenericDao);
-        managementMBean = new SdtManagementMBean();
+
+        ISdtManagementMBean managementMBean = new SdtManagementMBean();
         cache.setManagementMBean(managementMBean);

```

Comment:
Is it worth setting up a mock object for the SdtManagementMBean?  It's a separate component to the ErrorMessagesCache object so technically should be mocked.  Might've been left out because of the number of times it is invoked.  If so it'll be easier to set it up under Mockito.

Reply:
crated mock object         ISdtManagementMBean managementMBean = Mockito.mock(ISdtManagementMBean.class);

Reply:
Does the expected behaviour also need to be set and verified for the mocked management bean?  It's used by AbstractCacheControl.uncacheRequired() which is invoked from AbstractCacheControl.getValue() - all the tests in this class use getValue().

Reply:
verify(managementMBean).getCacheResetControl();

Reply:
Confirmed that SdtManagementMBean behaviour is verified by tests.